admin/class-boldgrid-backup-admin-restore-helper.php

```diff
@@ -237,39 +237,89 @@ public function prepare_restore() {
 	 */
 	public function set_writable_permissions( $archive_filepath ) {
-		global $wp_filesystem;
-
-		$zip = new ZipArchive();
-
-		if ( $zip->open( $archive_filepath ) ) {
-			for ( $i = 0; $i < $zip->numFiles; $i++ ) { // phpcs:ignore WordPress.NamingConventions.ValidVariableName
-				$data = $zip->statIndex( $i );
-
-				if ( empty( $data['name'] ) ) {
-					continue;
+		$is_compressor_type = $this->is_compressor_type();
+		if ( 'ZipArchive' === $is_compressor_type ) {
+			$zip = new ZipArchive();
+			if ( $zip->open( $archive_filepath ) ) {
+				for ( $i = 0; $i < $zip->numFiles; $i++ ) { // phpcs:ignore WordPress.NamingConventions.ValidVariableName
+					$data = $zip->statIndex( $i );
+
+					if ( empty( $data['name'] ) ) {
+						continue;
+					}
+
+					if ( $this->change_file_permissions( $data['name'] ) ) {
+						continue;
+					} else {
+						return false;
+					}
 				}
-
-				$full_path = ABSPATH . $data['name'];
-
-				// If the file does not exists, no need to check its permissions.
-				if ( ! $wp_filesystem->exists( $full_path ) ) {
-					continue;
-				}
-
-				if ( ! $wp_filesystem->chmod( $full_path ) ) {
-					$this->errors[] = sprintf(
-						// translators: 1 The path to a file that cannot be restored due to file permissions.
-						__( 'Permission denied. Unable to restore the following file: %1$s', 'boldgrid-backup' ),
-						$full_path
-					);
-					return false;
+			}
+		} elseif ( 'PclZip' === $is_compressor_type ) {
+			$zip = new PclZip( $archive_filepath );
+			if ( $zip->listContent() ) {
+				foreach ( $zip->listContent() as $file ) {
+
+					if ( empty( $file['filename'] ) ) {
+						continue;
+					}
+
+					if ( $this->change_file_permissions( $file['filename'] ) ) {
+						continue;
+					} else {
+						return false;
+					}
 				}
 			}
 		}
+		return true;
+	}
 
+	/**
+	 * Updates file permission per file
+	 *
+	 * This takes an individual filepath derived from the lists of files
+	 * in an archive, and uses wp_filesystem to set the correct permissions
+	 *
+	 * @since SINCEVERSION
+	 * @param string $file_name name of file to be changed.
+	 * @return bool True if permissions were able to be updated successfully.
+	 */
+	public function change_file_permissions( $file_name ) {
+		global $wp_filesystem;
+		$full_path = ABSPATH . $file_name;
+
+		// If the file does not exists, no need to check its permissions.
+		if ( ! $wp_filesystem->exists( $full_path ) ) {
+			return true;
+		}
+		if ( ! $wp_filesystem->chmod( $full_path ) ) {
+			$this->errors[] = sprintf(
+				// translators: 1 The path to a file that cannot be restored due to file permissions.
+				__( 'Permission denied. Unable to restore the following file: %1$s', 'boldgrid-backup' ),
+				$full_path
+			);
+			return false;
+		}
 		return true;
 	}
 
+	/**
+	 * Is Compressor Type
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @return string
+	 */
+	public function is_compressor_type() {
```

Comment:
This method should be `get_compressor_type` or something starting with `get_`.  When it starts with `is_`, I would expect the return to be a boolean.